packages/ui-kit/src/hooks/use-selected-path.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Custom hook that manages (for multi paths/languages components) the selected path state and provides a function to update it.
+ * The selected path is stored in local storage and can be updated by dispatching a custom event.
+ * The selected path is also reflected in the URL query string. If a query string activePath parameter is present on page load,
+ * then it overrides the local storage value.
+ * The reason for this double mechanism is that local storage event listeners are not triggered by the same window
+ * that updates the value. See https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event
+ *
+ * @returns An object containing the selectedPath state and the updateSelectedPath function.
+ */
+import { useState, useEffect } from 'react';
+
+const PATH_LOCAL_STORAGE_KEY = 'selected_path';
+const PATH_CHANGE_EVENT = 'selectedPathChange';
+
+const useSelectedPath = () => {
+  const [selectedPath, setSelectedPath] = useState<string | null>(null);
+
+  useEffect(() => {
+    // Read the initial value from local storage
+    const urlParams = new URLSearchParams(window.location.search);
+    const activePath = urlParams.get('activePath');
+    // if a query string activePath parameter is present, then override the local storage value
+    if (activePath) {
+      localStorage.setItem(PATH_LOCAL_STORAGE_KEY, activePath);
+    }
+    const initialValue = localStorage.getItem(PATH_LOCAL_STORAGE_KEY);
+    setSelectedPath(initialValue);
+
+    // Subscribe to changes in local storage or custom event
+    const handleStorageChange = (event: StorageEvent) => {
+      if (event.key === PATH_LOCAL_STORAGE_KEY) {
+        setSelectedPath(event.newValue?.toLowerCase() || '');
+      }
+    };
+
+    const handleCustomEvent = (event: CustomEvent<string>) => {
+      setSelectedPath(event.detail?.toLowerCase() || '');
+    };
+
+    window.addEventListener('storage', handleStorageChange);
+    window.addEventListener(
+      PATH_CHANGE_EVENT,
+      handleCustomEvent as EventListener
+    );
+
+    // Clean up the event listener
+    return () => {
+      window.removeEventListener('storage', handleStorageChange);
+      window.removeEventListener(
+        PATH_CHANGE_EVENT,
+        handleCustomEvent as EventListener
+      );
+    };
+  }, []);
+
+  const updateSelectedPath = (inputValue: string) => {
```

Comment:
We might be able to refactor this update  a bit by making 2 single responsibility branches of behaviour. They both grab the value from their distinct places.
1. input value branch
2. url branch

For me its unclear what behaviour is unique to both paths or does all behaviour happen for both input and url?

We could even break this function into 2 so we have 1 wrapper which handles taking input or value and normalizes it to a single value. Then that value is passed to the pure update functionality

Reply:
I'm not sure what you mean here @ColinRosati.

The `updateSelectedPath` is part of the hook public API and allows to programmatically update the current selected path. This means performing 3 independent tasks:
1.  Update local storage `active_path`
2. Update the url `activePath` get parameter
3. Dispatch a custom event to update any other existing multi path components on the page.

Maybe I'm not getting your point here, we can discuss it in a meeting if you want

Reply:
Thanks for the response. That makes sense, we need all actions to happen